openglyph/include/openglyph/assets/asset_loader.hpp

```diff
@@ -56,11 +57,50 @@ class AssetLoader
     std::unique_ptr<khepri::io::Stream> open_map(std::string_view name);
 
 private:
+    /**
+     * @brief Provides access to asset files from a single data source (physical or MegaFile).
+     *
+     * Each AssetLayer represents one data path.
+     *
+     * The class abstracts the file-loading logic by attempting to open files in the following
+     * order:
+     * 1. From the physical file system rooted at `data_path`.
+     * 2. From a MegaFile archive using `MegaFileSystem`.
+     *
+     */
+    class AssetLayer
```

Comment:
Nit: I think you can just declare this class here, and move its definition to the .cpp file. Saves some unnecessary contents in the header file.
You'll have to define a default destructor for AssetLoader in the .cpp as well, though. And maybe delete the copy/move constructors/operators.